mmv1/third_party/terraform/utils/node_config.go.erb

```diff
@@ -883,6 +929,39 @@ func expandLinuxNodeConfig(v interface{}) *container.LinuxNodeConfig {
 	}
 }
 
+func expandSoleTenantConfig(v interface{}) *container.SoleTenantConfig {
+	if v == nil {
+		return nil
+	}
+	ls := v.([]interface{})
+	if len(ls) == 0 {
+		return nil
+	}
+	cfg := ls[0].(map[string]interface{})
+	affinitiesRaw, ok := cfg["node_affinity"]
+	if !ok {
+		return nil
+	}
+	affinities := make([]*container.NodeAffinity, 0)
+	for _, v := range affinitiesRaw.(*schema.Set).List() {
+		na := v.(map[string]interface{})
+
+		values := make([]string, 0)
```

Comment:
nit: You can use `tpgresource.ConvertStringArr` here: https://github.com/hashicorp/terraform-provider-google-beta/blob/23d314625fb420d42b7453d7b11d49869e3d38e4/google-beta/tpgresource/utils.go#L181-L183